cpp/src/io/parquet/writer_impl.cu

```diff
@@ -1511,24 +1566,48 @@ void writer::impl::write(table_view const& table, std::vector<partition_info> co
   part_frag_offset.push_back(part_frag_offset.back() + num_frag_in_part.back());
 
   auto d_part_frag_offset = cudf::detail::make_device_uvector_async(
-    part_frag_offset, _stream, rmm::mr::get_current_device_resource());
+    part_frag_offset, stream, rmm::mr::get_current_device_resource());
   cudf::detail::hostdevice_2dvector<gpu::PageFragment> row_group_fragments(
-    num_columns, num_fragments, _stream);
+    num_columns, num_fragments, stream);
+
+  // Create table_device_view so that corresponding column_device_view data
+  // can be written into col_desc members
+  // These are unused but needs to be kept alive.
+  auto parent_column_table_device_view = table_device_view::create(single_streams_table, stream);
+  rmm::device_uvector<column_device_view> leaf_column_views(0, stream);
 
   if (num_fragments != 0) {
     // Move column info to device
-    col_desc.host_to_device(_stream);
+    col_desc.host_to_device(stream);
     leaf_column_views = create_leaf_column_device_views<gpu::parquet_column_device_view>(
-      col_desc, *parent_column_table_device_view, _stream);
+      col_desc, *parent_column_table_device_view, stream);
 
-    init_row_group_fragments(
-      row_group_fragments, col_desc, partitions, d_part_frag_offset, max_page_fragment_size);
+    init_row_group_fragments(row_group_fragments,
+                             col_desc,
+                             partitions,
+                             d_part_frag_offset,
+                             max_page_fragment_size,
+                             stream);
   }
 
-  std::vector<size_t> const global_rowgroup_base = _agg_meta->num_row_groups_per_file();
+  std::unique_ptr<aggregate_writer_metadata> agg_meta;
+  if (!curr_agg_meta) {
+    agg_meta = std::make_unique<aggregate_writer_metadata>(
+      partitions, kv_meta, this_table_schema, num_columns, stats_granularity);
+  } else {
+    agg_meta = std::make_unique<aggregate_writer_metadata>(*curr_agg_meta);
+
+    //    verify the user isn't passing mismatched tables
+    CUDF_EXPECTS(agg_meta->schema_matches(this_table_schema),
+                 "Mismatch in schema between multiple calls to write_chunk");
+
+    agg_meta->update_files(partitions);
```

Comment:
I am wondering if agg_meta gets updated, and then later we OOM, should agg_meta be rolled back or thrown away? Or is the code after this guaranteed to not OOM?

And apologies if I misunderstand the way this is working. I am probably missing something

Reply:
This `agg_meta` is not the writer's `agg_meta`. Instead, it is a copy from writer's `agg_meta`. The writer's `agg_meta` will be updated (assigned to this) after the processing step is done.

Reply:
Got it, thank you!